src/index.ts

```diff
@@ -13,7 +13,7 @@ import type { BeAnObject, IIndexArray, IndexOptions } from './types';
  * class ClassName {}
  * ```
  */
-export function index<T = BeAnObject>(fields: T, options?: IndexOptions<T>): ClassDecorator {
+export function index<T = BeAnObject>(fields: Partial<Record<keyof T, any> & BeAnObject>, options?: IndexOptions<T>): ClassDecorator {
```

Comment:
any specific reason on why `& BeAnObject> is there? shouldnt the Record be enough?

Reply:
> any specific reason on why `& BeAnObject` is there? shouldnt the Record be enough?

It's because of a ts issue I believe. There are more than one arguments of the index decorator that uses the generics type. In the case where the generics type is not specified, ts compiler tries to determine the type of the generics using the type of the arguments. Depending on which argument it chooses, there may be undesired errors at compile time. See the pic below for an example.
![屏幕截图(7)](https://user-images.githubusercontent.com/23728505/121015740-e8081900-c7cd-11eb-95e5-f0010d2ff5e3.png)
My solution is simply union the type of the first argument with `{[key:string]: any}`. Perhaps there are more clever ways that I am not thinking of ATM?

Reply:
i think this was because of also defining the `weights` property, which also uses `T`, but because it does not use `about` (from the first parameter) and also `T` was actually just *any* object and not something defined, typescript used the common properties between parameter 1 and `weigths` (from parameter 2)

i fixed this in my commit by adding the class as the properties definition (how it should be)